File: 242-ValidAnagram/validAnagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  if (s.length !== t.length) return false;

  let used  = new Int16Array(26);

  for(let i = 0; i < s.length; i++) {
    used[s.charCodeAt(i) - 97]++;
    used[t.charCodeAt(i) - 97]--;
  }

  for (let i = 0; i < used.length; i++) {
    if (used[i] !== 0) return false;
  }

  return true;
};


console.log(isAnagram("anagram", "nagaram"))

File: valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        counter = [0] * 26
        if len(s) != len(t): return False
        for i in range(len(s)):
            counter[ord(s[i]) - ord('a')] += 1
            counter[ord(t[i]) - ord('a')] -= 1

        for val in counter:
            if val != 0:
                return False
        return True

if __name__ in "__main__":
    s = Solution()
    assert s.isAnagram("anagram", "nagaram") == True
    assert s.isAnagram("rat", "car") == False